### PR TITLE
Some fixes and minor changes

### PR DESCRIPTION
**Pull request by tevosouza**
I made some minor changes and fixes on the API definition screen
- Added a new modal dialog to show validation errors to the user
- Added JSON and WSDL validation before save to fix a bug on updateDefinitionWithManagedEndpoint when API is published but the definition type is not correct. (e.g. the API definition is WSDL format but the definition type is set to SwaggerJSON)
- Changed the behavior of save button it is disabled when No Api Definition is selected.
- Fixed a bug that does not allow to change API definition type to old value after save.

Screen:
image

Known Issues
I can't found any way to validate YAML (a performative way without including some new libraries).  =(
Include support to translations on my new error dialog.

Feel free to suggest changes

PS: 
I'm sorry for the number of commits.
I made a mistake when doing rebase and lost some commits, so I had to redo it.

### CODE REVIEW

**Comment by volkflo** [comment]
Sounds good, I will have a look and try this probably next week :)

**Comment by EricWittmann** [comment]
Looks good to me, although some of the diffs are a little hard to follow.  I assume changes to tabbing levels or other unimportant formatting is causing it.  Also some of the code formatting doesn't match the rest of the codebase (e.g. brackets on separate lines) but I don't normally care much about that sort of thing as long as the formatting is still reasonable.  But others value formatting consistency more than I do, so I just wanted to mention it. :)

**Comment by tevosouza** [comment]
> Looks good to me, although some of the diffs are a little hard to follow. I assume changes to tabbing levels or other unimportant formatting is causing it. Also some of the code formatting doesn't match the rest of the codebase (e.g. brackets on separate lines) but I don't normally care much about that sort of thing as long as the formatting is still reasonable. But others value formatting consistency more than I do, so I just wanted to mention it. :)

Yep, I'm using vscode and it built-in formatters for *.ts and I'll take care of it when sending new PR's.

**Comment by volkflo** [comment]
@tevosouza I did a little bit of refactoring. Please check out: https://github.com/apiman/apiman/pull/1075

**Comment by tevosouza** [comment]
Replaced by #1075
